app/models/atmosphere/compute_site.rb

```diff
@@ -80,6 +80,8 @@ class ComputeSite < ActiveRecord::Base
               presence: true,
               inclusion: %w(openstack aws rackspace google_compute)
 
+    validate :nic_provider_class_defined, unless: "nic_provider_class_name.nil?"
```

Comment:
I think that if: :nic_provider_class_name is always true because it really is if: :symbol 
which evaluates always to true so tests are still green. But I am not 100% sure about it.